.github/workflows/ruby.yml

```diff
@@ -106,3 +106,31 @@ jobs:
       - run: bundle exec rake test:valgrind
         env:
           RUBY_FREE_AT_EXIT: 1
+  C99_compile:
```

Comment:
This job compiles the extension without C23 extensions to ensure it compiles with C99 compilers.

To compile it with `clang`, not `gcc`, the macos platform is used.